src/Microsoft.Azure.WebJobs.Host/Bindings/ExecutionContext/ExecutionContext.cs

```diff
@@ -32,5 +32,10 @@ public class ExecutionContext
         /// A host can set this via <see cref="CoreJobHostConfigurationExtensions.UseCore(JobHostConfiguration, string)"/>
         /// </remarks>
         public string FunctionAppDirectory { get; set; }
+
+        /// <summary>
+        /// Gets or sets value for retry count.
+        /// </summary>
+        public int RetryCount { get; set; }
```

Comment:
Issue https://github.com/Azure/azure-webjobs-sdk/issues/2595 mentions that the use case is to be able to check if RetryCount = MaxRetries. This doesn't enable that, right? Was it decided that that wasn't needed?

Reply:
I did not read #2595 carefully enough. I added `IRetryStratagy` to `RetryPolicy` and exposed 'RetryPolicy' in 'ExecutionContext'.

> However, the logic for determining whether this is truly the last retry would also have to also take the message dequeue count into account in the case of SB/Queues, since there are two levels of retries going on. E.g. if we're at the max dequeue count AND the max retry, we know this is the very last attempt.

We need to point out this in our docs. By design the retry logic only execute function code inside one instance and do not know about specific trigger retries mechanism(it's possible that another instance will retry execution for a storage/SB message).

Reply:
It is a little strange to add IRetryStrategy to RetryContext, because that context was designed to pass into IRetryStrategy.GetNextDelay. We're now wanting to pass RetryContext in other contexts and want more info on it. Another option would be to add MaxRetries to RetryContext. That is also a bit of duplication from the standpoint of an IRetryStrategy implementor, but at least leaves us with a single context object with all the info needed for inspection by other consumers (e.g. ExecutionContext). Thoughts @pragnagopa? I don't think IRetryStrategy itself is needed for ExecutionContext, since that interface is for determining the next delay, something that we wouldn't want function code calling since it's stateful.

Reply:
anyway, I added MaxRetriesCount to ExecutionContext  instead of IRetryStrategy. I think it fits better.